jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/RuleSetNodeInstance.java

```diff
@@ -75,6 +75,8 @@
     private static final Logger logger = LoggerFactory.getLogger(RuleSetNodeInstance.class);
     
     private static final String ACT_AS_WAIT_STATE_PROPERTY = "org.jbpm.rule.task.waitstate";
+    private static final String FIRE_RULE_LIMIT_PARAMETER = "FireRuleLimit";
+    private static final int DEFAULT_FIRE_RULE_LIMIT = Integer.parseInt(System.getProperty("org.jbpm.rule.task.firelimit", "100"));
```

Comment:
@mswiderski I think 100 is too low for a default limit. If the user does not define a limit, I think something like 10000 would be a better limit as it will not break existing cases, but still prevent infinite loops.

Reply:
here you go sir, 10000 it is :)